Makefile

```diff
@@ -27,8 +27,8 @@ TESTPARALLELISM := 4
 # We set the PLUGIN_VERSION to be the same as the version we use when building
 # the provider (e.g. x.y.z-dev-... instead of x.y.zdev...)
 build:: tfgen provider
-	for LANGUAGE in "nodejs" "python" "go" "dotnet" ; do \
-		$(TFGEN) $$LANGUAGE --overlays overlays/$$LANGUAGE/ --out ${PACKDIR}/$$LANGUAGE/ || exit 3 ; \
+	cd provider && for LANGUAGE in "nodejs" "python" "go" "dotnet" ; do \
+		$(TFGEN) $$LANGUAGE --overlays overlays/$$LANGUAGE/ --out ../${PACKDIR}/$$LANGUAGE/ || exit 3 ; \
```

Comment:
Can we clean this up at all?  Feels like the directory management is a bit "ad-hoc" now - though I don't have an immediately better suggestion.

Reply:
The correct way to do this is to have recursive make files then call `$(MAKE) -C provider` however we have an open item to revisit the entire Makefile anyway, so I'm deferring until the time we do that.